projects/pastanaga/src/lib/button/button-base.ts

```diff
@@ -1,34 +1,82 @@
-import {AfterContentInit, ElementRef, EventEmitter, HostBinding, Input, Output, SimpleChanges, ViewChild, ChangeDetectorRef, ViewRef} from '@angular/core';
+import {
+    AfterContentInit,
+    ChangeDetectorRef,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    Input,
+    Output,
+    ViewChild,
+    ViewRef
+} from '@angular/core';
 import { coerceBooleanProperty } from '@angular/cdk/coercion';
 
 const COLORS = ['primary', 'secondary', 'destructive', 'contrast'];
 const SIZES = ['tiny', 'small', 'large'];
 
 export class ButtonBase implements AfterContentInit {
-    @Input() color: 'primary'|'secondary'|'destructive'|'contrast' = 'primary';
-    @Input() size: 'tiny'|'small'|'large'|'' = '';
-    @Input() border = false;
-    @Input() disabled = false;
+    @Input() set color(value: 'primary' | 'secondary' | 'destructive' | 'contrast') {
+        if (!!value) {
+            COLORS.forEach(color => {
+                const colorClass = this.getClassFromInput('color', color, COLORS);
+                this.buttonStyle[colorClass] = color === value;
+            });
+        }
+    }
+
+    @Input() set size(value: 'tiny' | 'small' | 'large' | '') {
+        if (!!value) {
+            SIZES.forEach(size => {
```

Comment:
I know the button component is OnPush but if someday the base class is inherited by a non-OnPush, it would be much better to do:
```typescript
const buttonStyle: any = {...this.buttonStyle};
SIZES.forEach(size => {
                const sizeClass = this.getClassFromInput('size', size, SIZES);
                buttonStyle[sizeClass] = size === value;
            });
this.buttonStyle = buttonStyle;
```